### PR TITLE
Sign Up Page

### PR DESCRIPTION
**Pull request by nehalprasad**
@chanda6299797578 @chandack If you want i can code and show you how can we code.

### CODE REVIEW

**Comment by chandack** [comment]
I modified these two files first is  (src/show/component/TextField/index.js)  and another is  (src/show/pages/public/SignIn/index.js). 
 I didn’t create a pull request this time because it had already been done.

File: src/show/components/TextField/index.js [hunk 1]
```diff
@@ -1,23 +1,79 @@
...
+  onChange,
+  hasEye,
+}) => {
+  const [error, setError] = useState('');
+  const [inputType, setInputType] = useState(type);
+
+  const validate = (value) => {
+    if (type === 'email') {
+      const emailPattern = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+      if (!emailPattern.test(value)) {
```
Comment (nehalprasad) [inline src/show/components/TextField/index.js:17]: const validate = (type, value) => {
  const validationRules = {
    email: {
      pattern: /^[^\s@]+@[^\s@]+\.[^\s@]+$/,
      errorMessage: 'Invalid email address'
    },
    password: {
      pattern: /^(?=.*[0-9])(?=.*[!@#$%^&*])[A-Za-z\d!@#$%^&*]{8,}$/,
      errorMessage: 'Password must be at least 8 characters long and include at least one number and one special character'
    }
  };

  // Check if the type exists in the validation rules
  if (validationRules[type]) {
    const { pattern, errorMessage } = validationRules[type];
    if (!pattern.test(value)) {
      setError(errorMessage);
      return;
    }
  }

  setError(''); 
};

File: src/show/components/TextField/index.js [hunk 2]
```diff
@@ -1,23 +1,79 @@
...
+        className={`w-full border rounded-md text-xs p-1 ${error ? 'border-red-500' : 'border-gray-300'} focus:outline-none`}
+      />
+      {hasEye && type === 'password' && (
+        <button
+          type="button"
+          onClick={togglePasswordVisibility}
+          className="absolute pt-2 right-2"
+        >
+          {inputType === 'password' ? (
+            <svg className="h-2.5 w-2.5 text-gray-600" fill="none" stroke="currentColor" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
```
Comment (nehalprasad) [inline src/show/components/TextField/index.js:63]: use svg as component and save all this in assets under svg folder and create an index.js file there

File: src/show/components/TextField/index.js [hunk 3]
```diff
@@ -1,23 +1,79 @@
...
+            </svg>
+          ) : (
+            <svg className="h-2.5 w-2.5 text-gray-600" fill="none" stroke="currentColor" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
+              <path strokeLinecap="round" strokeLinejoin="round" strokeWidth="2" d="M15 12a3 3 0 00-6 0 3 3 0 006 0z"></path>
+              <path strokeLinecap="round" strokeLinejoin="round" strokeWidth="2" d="M2.003 12S5.003 5 12 5c6.997 0 9.997 7 9.997 7S18.997 19 12 19c-6.997 0-9.997-7-9.997-7z"></path>
+            </svg>
+          )}
+        </button>
+      )}
+      <p className="text-red-500 text-xs">{error}</p>
```
Comment (nehalprasad) [inline src/show/components/TextField/index.js:74]: it should be {error && <p className="text-red-500 text-xs">{error}</p>}

File: src/show/pages/public/SignIn/index.js [hunk 4]
```diff
@@ -1,13 +1,66 @@
...
+  const handleInputChange = (name, value) => {
+    setInputs((prev) => ({ ...prev, [name]: value }));
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+  };
+
+  const isFormValid = Object.values(errors).every((error) => error === '') &&
+    Object.values(inputs).every((value) => value !== '');
```
Comment (nehalprasad) [inline src/show/pages/public/SignIn/index.js:24]: const isFormValid = Object.entries(errors).every(([key, error]) => !error && inputs[key] !== '');

File: src/show/pages/public/SignIn/index.js [hunk 5]
```diff
@@ -1,13 +1,66 @@
...
+
   return (
-    <div className='flex justify-center items-center flex flex-row'>
-        <TextField label='Email' placeholder='Enter Email' required  />
-        <TextField label='Password' placeholder='Enter Password' required password />
+    <div className="flex justify-center mt-6">
+      <div className="w-64 bg-white">
+        <h2 className="font-bold mb-4" style={{fontSize: "1.4rem"}}>Sign in</h2>
```
Comment (nehalprasad) [inline src/show/pages/public/SignIn/index.js:29]: no inline styling

File: src/show/pages/public/SignIn/index.js [hunk 6]
```diff
@@ -1,13 +1,66 @@
...
+            value={inputs.password}
+            onChange={(value) => handleInputChange('password', value)}
+            required
+            hasEye={true}
+          />
+          <button
+            type="submit"
+            disabled={!isFormValid}
+            className={`w-full p-1.5 mt-4 text-white font-semibold rounded-full  ${isFormValid ? 'bg-blue-500' : 'bg-gray-500 cursor-pointer'}`}
+            style={{ fontSize: "0.6rem" }}
```
Comment (nehalprasad) [inline src/show/pages/public/SignIn/index.js:54]: no inline

File: src/show/pages/public/SignIn/index.js [hunk 7]
```diff
@@ -1,13 +1,66 @@
...
+          >
+            Sign in
+          </button>
+          <p className="mt-1 text-center font-semibold" style={{color: "#F72585", fontSize: "0.6rem"}}>Forgot Password?</p>
```
Comment (nehalprasad) [inline src/show/pages/public/SignIn/index.js:58]: no inline

File: src/show/pages/public/SignIn/index.js [hunk 8]
```diff
@@ -1,13 +1,66 @@
...
+            label="Password"
+            type="password"
+            name="password"
+            placeholder="Enter Password"
+            value={inputs.password}
+            onChange={(value) => handleInputChange('password', value)}
+            required
+            hasEye={true}
+          />
+          <button
```
Comment (nehalprasad) [inline src/show/pages/public/SignIn/index.js:50]: create a buttonComponent as well

File: src/show/components/TextField/index.js [hunk 9]
```diff
@@ -1,23 +1,79 @@
...
+  };
+
+  const togglePasswordVisibility = () => {
+    setInputType(inputType === 'password' ? 'text' : 'password');
+  };
+
+  return (
+    <div className="relative mb-4">
+      {label && (
+        <label className="block mb-1 font-medium text-zinc-400" style={{ fontSize: "0.6rem" }}>
```
[the remark this answers is not in the document]
Comment (chandack) [reply src/show/components/TextField/index.js]: removed

File: src/show/components/TextField/index.js (hunk 1 above)
Comment (chandack) [reply src/show/components/TextField/index.js]: i replaced previous code with this one

File: src/show/components/TextField/index.js (hunk 2 above)
Comment (chandack) [reply src/show/components/TextField/index.js]: done

File: src/show/components/TextField/index.js (hunk 3 above)
Comment (chandack) [reply src/show/components/TextField/index.js]: Done

File: src/show/pages/public/SignIn/index.js (hunk 4 above)
Comment (chandack) [reply src/show/pages/public/SignIn/index.js]: replaced

File: src/show/pages/public/SignIn/index.js (hunk 5 above)
Comment (chandack) [reply src/show/pages/public/SignIn/index.js]: Done

File: src/show/pages/public/SignIn/index.js (hunk 6 above)
Comment (chandack) [reply src/show/pages/public/SignIn/index.js]: Done

File: src/show/pages/public/SignIn/index.js (hunk 8 above)
Comment (chandack) [reply src/show/pages/public/SignIn/index.js]: Done

**Comment by chandack** [comment]
I have updated all the changes you mentioned. Please review it and let me know where I need to improve.

File: src/show/components/Button/index.js [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+  className = '',
+}) => {
+  return (
+    <button
+      type={type}
+      disabled={disabled || !isFormValid}
+      onClick={onClick}
+    //   className={`w-full p-1.5 mt-4 text-white font-semibold text-xs rounded-full ${
+    //     isFormValid ? 'bg-blue-500' : 'bg-gray-500 cursor-not-allowed'
+    //   }`}
```
Comment (nehalprasad) [inline src/show/components/Button/index.js:18]: remove commented code

File: src/show/components/Button/index.js [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+
+const Button = ({
+  type = 'button',
+  disabled = false,
+  isFormValid = true,
+  onClick,
+  children,
+  className = '',
+}) => {
+  return (
```
Comment (nehalprasad) [inline src/show/components/Button/index.js:11]: no return required

File: src/show/components/Button/index.js [hunk 12]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+
+const Button = ({
+  type = 'button',
```
Comment (nehalprasad) [inline src/show/components/Button/index.js:4]: see mui buttons or bootstrap button , where you can see type like 'Outline', Filled and others

File: src/show/components/Button/index.js [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+  disabled = false,
+  isFormValid = true,
+  onClick,
+  children,
```
Comment (nehalprasad) [inline src/show/components/Button/index.js:8]: give defaults to every props.

File: src/show/components/TextField/index.js [hunk 14]
```diff
@@ -1,23 +1,83 @@
-import React from "react";
-
+import React, { useState } from 'react';
+import { ReactComponent as EyeIcon } from '../../assets/svg/eye-icon.svg';
```
Comment (nehalprasad) [inline src/show/components/TextField/index.js:2]: Don't do this here. Create a index.js file inside the svg and 
import { ReactComponent as EyeIcon } from '../../assets/svg/eye-icon.svg'

export const icons =[ EyeIcon ]

icons.map(icon => {
  const Icon = props => {
    const ShowIcon = icon;
    return <ShowIcon {...props} />;
  };
  return Icon;
});

export {
EyeIcon
}

Comment (nehalprasad) [reply src/show/components/TextField/index.js:2]: same for all the svg, we will use in the future

File: src/show/components/TextField/index.js [hunk 15]
```diff
@@ -1,23 +1,83 @@
...
-      type={password ? "password" : "text"}
-      password={true}
-      className="bg-gray-100 text-black w-64 px-2 rounded border-2 border-black"
-    //   value={value}
-    />
-    <h6 className="text-xs text-right">{required && "required"}</h6>
-  </div>
-);
+  onChange,
+  hasEye,
```
Comment (nehalprasad) [inline src/show/components/TextField/index.js:10]: give defaults values to all props. if possible install this library and start giving types as well at the end something like this 

TextField.propTypes = {
name :  PropTypes.string,
type : PropTypes.oneOf[{'text', 'password'}],
onChange : PropTypes.func
}

export default TextField;

Comment (nehalprasad) [reply src/show/components/TextField/index.js:10]: this should be done in every functional component . if we create a component or we create a page . Defaults and proptypes is a good practise

File: src/show/components/TextField/index.js [hunk 16]
```diff
@@ -1,23 +1,83 @@
...
+    pattern: /^[^\s@]+@[^\s@]+.[^\s@]+$/,
+    errorMessage: 'Invalid email address'
+    },
+    password: {
+    pattern: /^(?=.[0-9])(?=.[!@#$%^&])[A-Za-z\d!@#$%^&]{8,}$/,
+    errorMessage: 'Password must be at least 8 characters long and include at least one number and one special character'
+    }
+    };
+    
+    // Check if the type exists in the validation rules
```
Comment (nehalprasad) [inline src/show/components/TextField/index.js:27]: remove comments . code yourself . try to minimize the usage of chatgpt

File: src/show/components/TextField/index.js [hunk 17]
```diff
@@ -1,23 +1,83 @@
...
+
+  const handleChange = (e) => {
+    const { value } = e.target;
+    validate(value);
+    onChange(value);
+  };
+
+  const togglePasswordVisibility = () => {
+    setInputType(inputType === 'password' ? 'text' : 'password');
+  };
```
Comment (nehalprasad) [inline src/show/components/TextField/index.js:47]: make one liner functions

File: src/show/components/TextField/index.js [hunk 18]
```diff
@@ -1,23 +1,83 @@
...
+
+  return (
+    <div className="relative mb-4">
```
Comment (nehalprasad) [inline src/show/components/TextField/index.js:50]: why relative . why we need position property

File: src/show/components/TextField/index.js [hunk 19]
```diff
@@ -1,23 +1,83 @@
...
+      )}
+      <input
+        type={inputType}
+        name={name}
+        placeholder={placeholder}
+        value={value}
+        onChange={handleChange}
+        className={`w-full border rounded-md text-xs p-1 ${error ? 'border-red-500' : 'border-gray-300'} focus:outline-none`}
+      />
+      {hasEye && type === 'password' && (
```
Comment (nehalprasad) [inline src/show/components/TextField/index.js:64]: why we need this extra hasEye props, if the type is given password, then it should automatically take that

File: src/show/components/TextField/index.js [hunk 20]
```diff
@@ -1,23 +1,83 @@
...
+        <button
+          type="button"
+          onClick={togglePasswordVisibility}
+          className="absolute pt-2 right-2"
+        >
+          {inputType === 'password' ? (
+            <EyeIcon className="h-2.5 w-2.5 text-gray-600" />
+          ) : (
+            <EyeIcon className="h-2.5 w-2.5 text-gray-600" />
```
Comment (nehalprasad) [inline src/show/components/TextField/index.js:73]: same eyeIcons component , what's the difference

File: src/show/components/TextField/index.js [hunk 21]
```diff
@@ -1,23 +1,83 @@
...
+          )}
+        </button>
+      )}
+      
+      <p className="text-red-500 text-xs">{error && error}</p>
```
Comment (nehalprasad) [inline src/show/components/TextField/index.js:78]: `{error &&<p className="text-red-500 text-xs"> {error}</p>}`

File: src/show/pages/public/SignIn/index.js [hunk 22]
```diff
@@ -1,13 +1,73 @@
-import React from 'react'
-import { TextField } from '../../../components'
 
-function SignIn() {
+import React, { useState } from 'react';
+import { TextField } from '../../../components'; 
+import Button from '../../../components/Button';
```
Comment (nehalprasad) [inline src/show/pages/public/SignIn/index.js:4]: Both Button and TextField are coming from the components, why the import is different. maintain the flow please

File: src/show/pages/public/SignIn/index.js [hunk 23]
```diff
@@ -1,13 +1,73 @@
...
+  const handleInputChange = (name, value) => {
+    setInputs((prev) => ({ ...prev, [name]: value }));
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+  };
+
+  // const isFormValid = Object.values(errors).every((error) => error === '') &&
+  //   Object.values(inputs).every((value) => value !== '');
```
Comment (nehalprasad) [inline src/show/pages/public/SignIn/index.js:25]: remove commented code

File: src/show/pages/public/SignIn/index.js [hunk 24]
```diff
@@ -1,13 +1,73 @@
...
+            onChange={(value) => handleInputChange('password', value)}
+            required
+            hasEye={true}
+          />
+          {/* <button
+            type="submit"
+            disabled={!isFormValid}
+            className={`w-full p-1.5 mt-4 text-white font-semibold text-xs rounded-full  ${isFormValid ? 'bg-blue-500' : 'bg-gray-500 cursor-pointer'}`}
+          >
+            Sign in
```
Comment (nehalprasad) [inline src/show/pages/public/SignIn/index.js:57]: please remove commented code. don't push commented code in future please

**Comment by chandack** [comment]
I have updated all the changes

File: src/show/components/Button/index.js [hunk 25]
```diff
@@ -0,0 +1,24 @@
...
+    type={type}
+    disabled={disabled || !isFormValid}
+    onClick={onClick}
+    className={`w-full p-1.5 mt-4 text-white font-semibold text-xs rounded-full ${
+      isFormValid ? 'bg-blue-500' : 'bg-gray-500 cursor-not-allowed'
+    } ${className}`}
+  >
+    {children}
+  </button>
+);
```
Comment (nehalprasad) [inline src/show/components/Button/index.js:22]: PropTypes missing here

File: src/show/components/Button/index.js [hunk 26]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+
+
+const Button = ({
+  type = 'button',
```
Comment (nehalprasad) [inline src/show/components/Button/index.js:5]: see mui buttons or bootstrap button , where you can see type like 'Outline', Filled and others

File: src/show/components/TextField/index.js [hunk 27]
```diff
@@ -1,23 +1,79 @@
-import React from "react";
-
+import React, { useState } from 'react';
+import { EyeIcon } from "../../assets/svg";
 const TextField = ({
+  name,
```
Comment (nehalprasad) [inline src/show/components/TextField/index.js:4]: default missing of some

File: src/show/components/TextField/index.js [hunk 28]
```diff
@@ -1,23 +1,79 @@
...
-      className="bg-gray-100 text-black w-64 px-2 rounded border-2 border-black"
-    //   value={value}
-    />
-    <h6 className="text-xs text-right">{required && "required"}</h6>
-  </div>
-);
-
-export default TextField;
+  onChange,
+  hasEye,
```
Comment (nehalprasad) [inline src/show/components/TextField/index.js:10]: why hasEye required when you can work with type password

**Comment by chandack** [comment]
https://meet.google.com/vid-pkjv-zdq

File: src/show/components/TextField/index.js (hunk 28 above)
Comment (chandack) [reply src/show/components/TextField/index.js]: done

File: src/show/components/TextField/index.js (hunk 27 above)
Comment (chandack) [reply src/show/components/TextField/index.js]: done

File: src/show/components/Button/index.js (hunk 26 above)
Comment (chandack) [reply src/show/components/Button/index.js:5]: sir I saw that but i did not get exactly how to do here.

**Comment by chandack** [comment]
sir one more thing i added  image in split component,  but i am doing some mistake there please review that and correct me.

File: src/show/components/Split/index.js [hunk 29]
```diff
@@ -0,0 +1,22 @@
+import PropTypes from "prop-types";
+import React, { Fragment } from "react";
+import LeftImage  from "../../assets/img/image.jpg";
+
+const Split = ({ leftContent = Fragment, rightContent = Fragment }) => {
+  return (
+    <div className="h-screen flex">
+      <div className="w-full h-full">{leftContent}
+      <img src={LeftImage} alt="Left side" className="object-cover" />
```
Comment (nehalprasad) [inline src/show/components/Split/index.js:9]: use Split on the SIgnIn and do something like this
 ```
<Split leftContent = {<img src={LeftImage} alt="Left side" className="object-cover" />}
rightContent={ <div className="flex justify-center mt-6">
      <div className="w-64 bg-white">
        <h2 className=" text-2xl font-bold mb-4">Sign in</h2>
        <form onSubmit={handleSubmit} className="signin-form">
          <TextField
            label="Email"
            type="email"
            name="email"
            placeholder="Enter Email"
            value={inputs.email}
            onChange={(value) => handleInputChange('email', value)}
            required
          />
          <TextField
            label="Password"
            type="password"
            name="password"
            placeholder="Enter Password"
            value={inputs.password}
            onChange={(value) => handleInputChange('password', value)}
            required
          />
          <Button
            type="submit"
            isFormValid={isFormValid}
          >
            Sign in
          </Button>
          <p className="mt-1 text-center font-semibold text-xs text-pink">Forgot Password?</p>
        </form>
      </div>
    </div>}
/>
```
this way we can pass different images here for different screen . remove the img from here

File: src/show/components/TextField/index.js [hunk 30]
```diff
@@ -1,23 +1,78 @@
...
+        pattern: /^[^\s@]+@[^\s@]+.[^\s@]+$/,
+        errorMessage: 'Invalid email address'
+      },
+      password: {
+        pattern: /^(?=.[0-9])(?=.[!@#$%^&])[A-Za-z\d!@#$%^&]{8,}$/,
+        errorMessage: 'Password must be at least 8 characters long and include at least one number and one special character'
+      }
+    };
+
+    // Check if the type exists in the validation rules
```
Comment (nehalprasad) [inline src/show/components/TextField/index.js:26]: remove commented code

File: src/show/components/TextField/index.js [hunk 31]
```diff
@@ -1,23 +1,78 @@
...
-  </div>
-);
-
-export default TextField;
+  onChange = () => {},
+}) => {
+  const [error, setError] = useState(''); 
+  const [inputType, setInputType] = useState(type);
+
+  const validate = (type, value) => {
```
Comment (nehalprasad) [inline src/show/components/TextField/index.js:14]: move this function outside of the TextField

File: src/show/components/TextField/index.js [hunk 32]
```diff
@@ -1,23 +1,78 @@
...
+      if (!pattern.test(value)) {
+        setError(errorMessage);
+        return;
+      }
+    }
+
+    setError('');
+  };
+
+  const handleChange = (e) => {
```
Comment (nehalprasad) [inline src/show/components/TextField/index.js:38]: remove it to the outside

File: src/show/components/TextField/index.js [hunk 33]
```diff
@@ -1,23 +1,78 @@
...
+    const { value } = e.target;
+    validate(value);
+    onChange(value);
+  };
+
+  const togglePasswordVisibility = () => {
```
Comment (nehalprasad) [inline src/show/components/TextField/index.js:44]: move outside of the TextField

File: src/show/components/TextField/index.js [hunk 34]
```diff
@@ -1,23 +1,78 @@
...
+    setInputType(inputType === 'password' ? 'text' : 'password');
+  };
+
+  return (
+    <div className="relative mb-4">
```
Comment (nehalprasad) [inline src/show/components/TextField/index.js:49]: why relative?

File: src/show/pages/public/SignIn/index.js [hunk 35]
```diff
@@ -1,13 +1,62 @@
...
+  const [inputs, setInputs] = useState({
+    email: '',
+    password: '',
+  });
+  const [errors] = useState({
+    email: '',
+    password: '',
+  }); 
+
+  const handleInputChange = (name, value) => {
```
Comment (nehalprasad) [inline src/show/pages/public/SignIn/index.js:15]: move out of the main function

File: src/show/pages/public/SignIn/index.js [hunk 36]
```diff
@@ -1,13 +1,62 @@
...
+    setInputs((prev) => ({ ...prev, [name]: value }));
+  };
+
+  const handleSubmit = (e) => {
```
Comment (nehalprasad) [inline src/show/pages/public/SignIn/index.js:19]: move out of the main function

File: src/show/components/TextField/index.js (hunk 33 above)
Comment (chandack) [reply src/show/components/TextField/index.js]: done

File: src/show/components/TextField/index.js (hunk 32 above)
Comment (chandack) [reply src/show/components/TextField/index.js]: done

File: src/show/components/TextField/index.js (hunk 31 above)
Comment (chandack) [reply src/show/components/TextField/index.js]: when i moved this function outside of the  textfield,  i got an error   "useState" cannot be called at the top level. then how to move outside.

File: src/show/components/TextField/index.js (hunk 30 above)
Comment (chandack) [reply src/show/components/TextField/index.js]: done

File: src/show/pages/public/SignIn/index.js (hunk 35 above)
Comment (chandack) [reply src/show/pages/public/SignIn/index.js]: done

File: src/show/components/Split/index.js (hunk 29 above)
Comment (chandack) [reply src/show/components/Split/index.js]: done

File: src/show/components/TextField/index.js (hunk 34 above)
Comment (chandack) [reply src/show/components/TextField/index.js:49]: icon should stay inside the placeholder

File: src/show/components/TextField/index.js [hunk 37]
```diff
@@ -1,23 +1,77 @@
...
+  const { value } = e.target;
+  validate(e.target.type, value, setError);
+  onChange(value);
+};
+
+const togglePasswordVisibility = (inputType, setInputType) => {
+  setInputType(inputType === 'password' ? 'text' : 'password');
+};
+//  const [error, setError] = useState('');
+//   const [inputType, setInputType] = useState(type);
```
Comment (nehalprasad) [inline src/show/components/TextField/index.js:35]: commented code

**Pull request merged**